aspnet/data/entity-framework-6.rst

```diff
@@ -0,0 +1,136 @@
+
+Code-based configuration is achieved by creating a subclass of ``System.Data.Entity.Config.DbConfiguration`` and applying ``System.Data.Entity.DbConfigurationTypeAttribute`` to your ``DbContext`` subclass.
+
+Our config file typically looked like this:
+
+.. code-block:: xml
+    :linenos:
+    
+    <entityFramework>
+        <defaultConnectionFactory type="System.Data.Entity.Infrastructure.LocalDbConnectionFactory, EntityFramework">
+            <parameters>
+                <parameter value="mssqllocaldb" />
+            </parameters>
+        </defaultConnectionFactory>
+        <providers>
+            <provider invariantName="System.Data.SqlClient" type="System.Data.Entity.SqlServer.SqlProviderServices, EntityFramework.SqlServer" />
+        </providers>
+    </entityFramework>
+
+The ``defaultConnectionFactory`` element sets the factory for connections. If this attribute is not set then the default value is ``SqlConnectionProvider``. If, on the other hand, value is provided, the given class will be used to create ``DbConnection`` with its ``CreateConnection`` method. If the given factory has no default constructor then you must add parameters that are used to construct the object.
+
+.. code-block:: c#
+    :linenos:
+
+    [DbConfigurationType(typeof(CodeConfig))] // point to the class that inherit from DbConfiguration
+    public class ApplicationDbContext : DbContext
+    {
+        [...]
+    }
+    
+    public class CodeConfig : DbConfiguration
+    {
+        public CodeConfig()
+        {
+            SetDefaultConnectionFactory(new System.Data.Entity.Infrastructure.SqlConnectionFactory());
+            SetProviderServices("System.Data.SqlClient",
+                System.Data.Entity.SqlServer.SqlProviderServices.Instance);
+            }
+        }
+    }
+    
+SQL Server and SQL Server Express
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the default and so no explicit configuration is needed. Optionally the above ``CodeConfig`` class can be used to explicitly set the connection factory and provider services.
+
+SQL Server Express LocalDB
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The below code configures code first to utilize local db.
+
+.. code-block:: c#
+    :linenos:
+    
+    [DbConfigurationType(typeof(CodeConfig))] // point to the class that inherit from DbConfiguration
+    public class ApplicationDbContext : DbContext
+    {
+        [...]
+    }
+    
+    public class CodeConfig : DbConfiguration
+    {
+        public CodeConfig()
+        {
+            SetDefaultConnectionFactory(new System.Data.Entity.Infrastructure.LocalDbConnectionFactory("mssqllocaldb"));
+            SetProviderServices("System.Data.SqlClient",
+                System.Data.Entity.SqlServer.SqlProviderServices.Instance);
+        }
+    }
+
+Notes on Automatic Migrations
+-----------------------------
```

Comment:
You should remove "Automatic" from here... the note provided is about migrations in general and not just automatic migrations.